src/gsy_e/models/strategy/forward/__init__.py

```diff
@@ -94,14 +119,122 @@ def event_offer_traded(self, *, market_id, trade):
-        self._post_orders_to_new_markets()
+        if self.fully_automated_trading:
+            self._post_orders_to_new_markets()
 
     @property
     def state(self) -> "StateInterface":
         """Get the state class of the strategy. Needs to be implemented by all strategies"""
         raise NotImplementedError
+
+    def apply_live_event(self, event: Dict):
+        """Apply the incoming live event to the strategy."""
+        self._live_event_handler.dispatch(event)
+
+
+class ForwardLiveEvents:
+    """Handle live events for the forward strategies."""
+
+    def __init__(self, strategy):
+        self._strategy = strategy
+
+    def dispatch(self, event: Dict):
+        """Apply a live event to the strategy."""
+        if not B2BLiveEvents.is_supported_event(event.get("type")):
+            self._strategy.log.error(
+                "Invalid event (%s) for area %s.", event, self._strategy.area.name)
+            return
+
+        if event.get("type") == B2BLiveEvents.ENABLE_TRADING_EVENT_NAME:
+            self._auto_trading_event(event.get("args"))
+        if event.get("type") == B2BLiveEvents.DISABLE_TRADING_EVENT_NAME:
+            self._stop_auto_trading_event(event.get("args"))
+        if event.get("type") == B2BLiveEvents.POST_ORDER_EVENT_NAME:
+            self._post_order_event(event.get("args"))
+        if event.get("type") == B2BLiveEvents.REMOVE_ORDER_EVENT_NAME:
+            self._remove_order_event(event.get("args"))
+
+    def _auto_trading_event(self, args):
+        """
+        Enable energy trading, and create order updater for all open markets between the start and
+        end time.
+        """
+        if not LiveEventArgsValidator(
+                self._strategy.log.error).validate_start_trading_event_args(args):
+            return
+
+        market_type = AvailableMarketTypes(args["market_type"])
+        start_time = str_to_pendulum_datetime(args["start_time"])
+        end_time = str_to_pendulum_datetime(args["end_time"])
+        capacity_percent = args["capacity_percent"]
+        energy_rate = args["energy_rate"]
+        market = self._strategy.area.forward_markets[market_type]
+
+        for slot in market.market_time_slots:
+            if not start_time <= slot <= end_time:
+                continue
+
+            if market not in self._strategy._order_updaters:
+                self._strategy._order_updaters[market] = {}
+            updater = self._strategy._order_updaters[market].get(slot)
+            if updater:
+                market.remove_open_orders(slot)
+
+            market_parameters = market.get_market_parameters_for_market_slot(slot)
+
+            order_updater_params = deepcopy(self._strategy._order_updater_params[market_type])
+            order_updater_params.capacity_percent = capacity_percent
+            order_updater_params.final_rate = energy_rate
+            order_updater_params.initial_rate = min(
+                order_updater_params.initial_rate, energy_rate)
```

Comment:
Doesn't this depend on whether you have a PV or a Load?

Reply:
Great catch! In order to not split the flow with if blocks for producer and consumer strategies, I will clamp the value between initial and final rate. That way it will never fall out of bounds of the predefined OrderUpdaterParams for each market type. Thanks again!

Reply:
Great, but don't we also clamp the final rate, too?

Reply:
The thing is that the final rate is configurable by the user. If we clamp it, then we will only allow the exchange to operate in a specific energy rate range, which is not something that we want. We only want to limit the starting price, not violate the desire of the user.

Reply:
Ok, so the `energy_rate` is rather a `not_buy_higher`/`not_sell_lower`- `rate` ?
Who is deciding about the initial values of the initial and final rates in the workflow. They are set when creating the setup AFAIS, right? So the owner of a PV power plant has to provide a bidding range when they join the exchange, right?
What if the like to exceed the initial/final range in the auto or manual trading mode? They just can not atm, right? Is there a motivation behind this that I do not see?

Reply:
First of all, this is only applicable for the auto trading mode, since the manual trading mode dictates a specific price and no updates then there is no concept of initial / final update prices. WRT auto mode, there is only one field in the UI actually, only to denote the final price that is acceptable for the customer. The initial price is left uncontrolled, therefore the customer does not define the initial / final range, only the upper / lower rate limit that he is willing to accept. The initial rate is defined by us, and at the moment is set when creating the setup (will typically have the default value from the OrderUpdaterParameter class). 
Let's try to rethink this using all possible cases. For instance, think of these values for a load strategy:
```
initial_rate = 10
final_rate = 30
energy_rate = 50
```
In this case, the correct behavior is to not change the energy rate of the user, but instead to move only the final rate to 50. Therefore the values will be:
```
initial_rate = 10
final_rate = 30
energy_rate = 50
```
Second cornercase scenario is this:
```
energy_rate = 5
initial_rate = 10
final_rate = 30
```
In this case the final_rate will be equal to energy_rate, and the initial_rate should again drop to 5:
```
energy_rate = 5
initial_rate = 5
final_rate = 5
```
With the clamping, this one will not work as far as I see (the original implementation would, but only for the load and not the PV). Similar observations occur for the PV too. Let me adapt the code in order to handle both cases. Thanks for insisting here!

Reply:
Thanks a lot for the verbose answer! It and the implementation makes sense now. 

>  The initial rate is defined by us,

So how would we find numbers that are suitable for "intelligent" trading and also allow a wide range for the user?
This is no question for this PR, but for a general discussion when we revise this feature. So, feel free to skip answering it. :-) 
Thanks again!